src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -386,21 +396,33 @@ void icm426xxGyroInit(gyroDev_t *gyro)
         gyro->gyroRateKHz = GYRO_RATE_1_kHz;
     }
 
+#if defined(USE_ACCGYRO_IIM42653)
+    STATIC_ASSERT(INV_FSR_4000DPS == 4, "INV_FSR_4000DPS must be 4 to generate correct value");
+    spiWriteReg(dev, ICM426XX_RA_GYRO_CONFIG0, (4 - INV_FSR_4000DPS) << 5 | (odrConfig & 0x0F));
```

Comment:
This branch will be used even for gyros with 2k dps/16G range. In both cases 0 is stored, so selected value is correct, but code pretends that INV_FSR_4000DPS is selected ... 

IMO best way now is to write `0 << 5` (in both cases) and add comment about actual rates selected.

Reply:
Agreed it was confusing. I've updated the code and added a note. Let me know if this is what you were thinking